packages/bff/src/services/agreementService.ts

```diff
@@ -1,7 +1,102 @@
 /* eslint-disable functional/immutable-data */
-import { agreementApi, catalogApi } from "pagopa-interop-api-clients";
-import { AgreementProcessClient } from "../providers/clientProvider.js";
-import { Headers } from "../utilities/context.js";
+/* eslint-disable @typescript-eslint/explicit-function-return-type */
+
+import { WithLogger } from "pagopa-interop-commons";
+import {
+  bffApi,
+  catalogApi,
+  agreementApi,
+  tenantApi,
+  attributeRegistryApi,
+} from "pagopa-interop-api-clients";
+import {
+  AgreementProcessClient,
+  PagoPAInteropBeClients,
+} from "../providers/clientProvider.js";
+import { BffAppContext, Headers } from "../utilities/context.js";
+import { agreementDescriptorNotFound } from "../model/domain/errors.js";
+import {
+  toCompactEservice,
+  toCompactDescriptor,
+} from "../model/api/apiConverter.js";
+import { getBulkAttributes } from "./attributeService.js";
+import { enhanceTenantAttributes } from "./tenantService.js";
+
+export function agreementServiceBuilder(clients: PagoPAInteropBeClients) {
+  const { agreementProcessClient } = clients;
+  return {
+    async createAgreement(
+      payload: bffApi.AgreementPayload,
+      { headers, logger }: WithLogger<BffAppContext>
+    ) {
+      logger.info(`Creating agreement with seed ${JSON.stringify(payload)}`);
+      return await agreementProcessClient.createAgreement(payload, {
+        headers,
+      });
+    },
+
+    async getAgreements({
+      offset,
+      limit,
+      producersIds,
+      eservicesIds,
+      consumersIds,
+      states,
+      ctx,
```

Comment:
if it's possible I'd like to isolate the context from the other filter parameter

Reply:
Yes i think it's a very good idea, it is semantically different from the other 👍🏻